lib/ficus.rb

```
@@ -1,7 +1,167 @@
 require 'yaml'
-require 'logger'
-require 'recursive-open-struct'
 
-require 'ficus/loader'
-require 'ficus/dsl'
-require 'ficus/exceptions'
+class Ficus
+  class << self
+    def load_file(filename)
+      Ficus.load(YAML.load_file(filename))
+    end
+
+    def load(data)
+      Ficus.new data
+    end
+  end
+
+  attr_accessor :config, :errors, :opts, :schema, :templates
+
+  TYPES = {
+    string:  [String],
+    number:  [Fixnum, Float],
+    boolean: [TrueClass, FalseClass],
+    array:   [Array],
+    section: [Hash],
+  }
+
+  def initialize(config, opts = {})
+    self.config = config
+    self.errors = []
+    self.opts = opts
+    self.templates = {}
+  end
+
+  def validation(&block)
+    self.schema = block
+  end
+
+  def valid?
+    self.errors = []
+    instance_eval(&self.schema) unless self.schema.nil?
```

Do we want to raise an error if self.schema is nil? might be worth a warning if somebody tries to run a nil validation.